### PR TITLE
Corrected index_col arg of read_csv to be a 5-way Union type.

### PR DESCRIPTION
**Pull request by MylesBartlett**

### CODE REVIEW

File: pandas-stubs/__init__.pyi [hunk 1]
```diff
@@ -32,7 +32,7 @@ def read_csv(
     filepath_or_buffer: Union[str, Path],
     sep: str = ...,
     header: Optional[int] = ...,
-    index_col: Optional[int] = ...,
+    index_col: Optional[Union[str, int, Sequence[str], Sequence[int], bool]] = ...,
```
Comment (thomkeh) [inline pandas-stubs/__init__.pyi:35]: don't use `Sequence[str]`

Comment (thomkeh) [inline pandas-stubs/__init__.pyi:35]: never use `Sequence[str]`

**Comment by thomkeh** [comment]
also, it recommended to use imperative voice in commit messages

File: pandas-stubs/__init__.pyi (hunk 1 above)
Comment (MylesBartlett) [reply pandas-stubs/__init__.pyi:35]: Couldn't you have made this a suggestion?

Comment (thomkeh) [inline pandas-stubs/__init__.pyi:35]:
```suggestion
    index_col: Optional[Union[str, int, List[str], Tuple[str, ...], Sequence[int], bool]] = ...,
```

Comment (MylesBartlett) [inline pandas-stubs/__init__.pyi:35]:
```suggestion
    index_col: Optional[Union[str, int, Union[List[str], Tuple[str, ...]], Sequence[int], bool]] = ...,
```

**Comment by thomkeh** [comment]
you have a nested Union now

**Pull request merged**